File: 1954-replace-all-digits-with-characters/replace-all-digits-with-characters.py
class Solution(object):
    def replaceDigits(self, s):
        """
        :type s: str
        :rtype: str
        """
        lastLetter=0
        st=""
        for x in s:
            if x.isalpha():
                st+=x
                lastLetter=x
            else:
                st+=chr(ord(lastLetter)+int(x))

        return st
        

File: 1954-replace-all-digits-with-characters/replace-all-digits-with-characters.java
class Solution {
    public String replaceDigits(String s) {
        StringBuffer sb=new StringBuffer();
        char lastLetter=0;
        for(char c:s.toCharArray()){
            if(Character.isLetter(c)){
                sb.append(c);
                lastLetter=c;
            }
            else{
                sb.append((char)((c-'0')+lastLetter));
            }
        }
        return sb.toString();
    }
}